Library/Homebrew/keg_relocate.rb

```diff
@@ -86,17 +86,24 @@ def change_rpath(file, new_prefix)
       return
     end
     return unless patchelf.installed?
-    cmd = "#{patchelf.bin}/patchelf --set-rpath #{new_prefix}/lib"
-    (old_i, old_p) = `#{patchelf.bin}/patchelf --print-rpath --print-interpreter #{file}`.split("\n")
+    old_path = `#{patchelf.bin}/patchelf --print-rpath #{file}`.strip
+    old_interp = ""
     raise ErrorDuringExecution.new(cmd) unless $?.success?
+    lib_path = "#{new_prefix}/lib"
+    rpath = old_path.split(":").map { |x| x.sub(old_prefix, new_prefix) }
```

Comment:
I want to reject paths that don't contain `old_prefix` to reject paths that are in for example either the host's libraries or in the build directory (untested below)
```ruby
rpath = old_path.split(":").select { |x| x.include? old_prefix }.map { |x| x.sub(old_prefix, new_prefix) }
```